### PR TITLE
Add tests and profiles for testing EAP6, SpringBoot and Fuse adapters

### LINKED ISSUE #20953: Add tests and profiles for testing Keycloak with EAP6, SpringBoot and Fuse adapters
### Description

Add tests and profiles for testing Keycloak with EAP6, SpringBoot and Fuse Keycloak 18.x adapters

### PR DESCRIPTION
**Pull request by pdrozd**
Closes #20953

### CODE REVIEW

**Comment by pdrozd** [comment]
This PR facilitates the testing of EAP, SpringBoot, and Fuse adapters that are not part of our current release. These tests aim to certify the previously released adapters, which are built on Keycloak 18, with the latest Keycloak version. In addition, this PR reintroduces previously removed profiles for Fuse and EAP, along with enhancements to enable testing of different adapter and server versions.

@mhajas @mabartos @miquelsi Can I ask you for a review?

**Comment by pdrozd** [comment]
Adapter pipeline successfully passed: https://jenkins.com/job/universal-test-pipeline-adapters/1700/

File: testsuite/integration-arquillian/test-apps/fuse/camel/pom.xml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        </keycloak.osgi.export>
+        <keycloak.osgi.import>
+            javax.servlet;version="[3.1,5)",
+            javax.servlet.http;version="[3.1,5)",
+            org.apache.camel.*,
+            org.apache.camel;version="[2.13,3)",
+            org.eclipse.jetty.security;version="[8,10)",
+            org.eclipse.jetty.server.nio;version="[8,10)",
+            org.eclipse.jetty.util.security;version="[8,10)",
+            org.keycloak.*;version="${project.version}",
```
Comment (mhajas) [inline testsuite/integration-arquillian/test-apps/fuse/camel/pom.xml:46]: As far as I understand, we want this for backward compatibility of older adapters, should we change this to something like here? https://github.com/pdrozd/keycloak/blob/adapters-18x/testsuite/integration-arquillian/test-apps/spring-boot-adapter-app/pom.xml#L109 so we can change the version?

File: testsuite/integration-arquillian/tests/other/springboot-tests/pom.xml [hunk 2]
```diff
@@ -16,11 +16,11 @@
 
         <maven.settings.file/>
         <springboot.version.option>2.4</springboot.version.option>
+        <spring.boot.adapter.version>${project.version}</spring.boot.adapter.version>
```
Comment (mhajas) [inline testsuite/integration-arquillian/tests/other/springboot-tests/pom.xml:19]: Why this defaults to `${project.version}`? In the description there is we want to test older adapters.

**Review by mhajas — COMMENTED** [review]
Thank you for PR @pdrozd 

It is quite big and hard to review. But I guess it is ok since it passed the pipeline. I have a few questions.

Do we want to run this within Github Actions? 

Should we also update the pipeline to support setting the adapter version for all adapter tests and probably check whether the version is correctly picked up?

File: testsuite/integration-arquillian/test-apps/fuse/camel/pom.xml (hunk 1 above)
Comment (pdrozd) [reply testsuite/integration-arquillian/test-apps/fuse/camel/pom.xml:46]: Good catch Michal, thanks!

File: testsuite/integration-arquillian/tests/other/springboot-tests/pom.xml (hunk 2 above)
Comment (pdrozd) [reply testsuite/integration-arquillian/tests/other/springboot-tests/pom.xml:19]: I would keep it as it is even if the intention is to test older versions of the adapter. There is still a springboot adapter for current releases, so it should be ok

**Comment by pdrozd** [comment]
@mhajas Thanks for you comments, I've changed the springboot adapter version in camel example. Currently these tests will not be executed within GHA. Regarding the pipeline, you are right, currently there is separate pipeline for testing the old version of the adapters. Once it will be stable enough it will be part of nightly run for adapter tests (jenkins)

File: testsuite/integration-arquillian/test-apps/fuse/camel-fuse7-undertow/pom.xml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        <keycloak.osgi.export>
+        </keycloak.osgi.export>
+        <keycloak.osgi.import>
+            javax.servlet;version="[3.1,5)",
+            javax.servlet.http;version="[3.1,5)",
+            javax.net.ssl,
+            org.apache.camel.*,
+            io.undertow.*;version="[1.4,3)",
+            org.apache.camel;version="[2.13,3)",
+            org.keycloak.*;version="${project.version}",
```
Comment (mhajas) [inline testsuite/integration-arquillian/test-apps/fuse/camel-fuse7-undertow/pom.xml:44]:
```suggestion
            org.keycloak.*;version="${fuse.adapter.version}",
```

**Review by mhajas — CHANGES_REQUESTED** [review]
@pdrozd the Github UI is quite unresponsive so I am not able to add comments to all places where it is needed, but I see more occurrences in test-apps where we need to change `project.version` to `fuse.adapter.version`. If you want we can jump on a call and have a look together.

File: testsuite/integration-arquillian/test-apps/fuse/customer-app-fuse/pom.xml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            org.apache.http.conn.*;version=${apache.httpcomponents.fuse.version},
+            org.apache.http.cookie.*;version=${apache.httpcomponents.fuse.version},
+            org.apache.http.impl.auth.*;version=${apache.httpcomponents.fuse.version},
+            org.apache.http.impl.client.*;version=${apache.httpcomponents.fuse.version},
+            org.apache.http.impl.conn.*;version=${apache.httpcomponents.fuse.version},
+            org.apache.http.impl.cookie.*;version=${apache.httpcomponents.fuse.version},
+            org.apache.http.impl.execchain.*;version=${apache.httpcomponents.fuse.version},
+            org.apache.http.*;version=${apache.httpcomponents.httpcore.fuse.version},
+            javax.servlet.*;version="[3.1,5)",
+            org.keycloak.adapters.authentication;version="${project.version}";resolution:=optional,
```
[the remark this answers is not in the document]
Comment (pdrozd) [reply testsuite/integration-arquillian/test-apps/fuse/customer-app-fuse/pom.xml:49]: Thanks Michal, let me do a deeper review :)

**Comment by pdrozd** [comment]
@mhajas I've updated correctly the versions for Fuse apps and retested

**Review by mhajas — APPROVED** [review]
Thank you @pdrozd, the code looks good to me, let's wait for the pipeline results and then merge.

**Review by mabartos — APPROVED** [review]
@pdrozd LGTM as well. 

Failures for Operator CI are not related to this PR.

**Review by miquelsi — APPROVED** [review]
Great job @pdrozd, thanks!

**Pull request merged**